addons/common/fnc_addWeaponWithoutItems.sqf

```diff
@@ -3,31 +3,33 @@
 Function: CBA_fnc_addWeaponWithoutItems
 
 Description:
-    Adds weapon to unit without attachments and without taking a magazine.
+    Adds weapon to unit without taking a magazine. Attachments will be removed by default, but can be kept by setting a parameter.
 
     Does not work on vehicles.
     Attempts to keep magazine ids for unrelated magazines.
 
 Parameters:
-    _unit   - Unit to add the weapon to <OBEJCT>
-    _weapon - Weapon to add <STRING>
+    _unit               - Unit to add the weapon to <OBEJCT>
+    _weapon             - Weapon to add <STRING>
+    _removeLinkedItems  - If linked items should be removed or not <BOOLEAN> (Default: true)
 
 Returns:
     Nothing.
 
 Examples:
     (begin example)
-        [player, "arifle_mx_F"] CBA_fnc_addWeaponWithoutItems;
+        [player, "arifle_mx_F"] call CBA_fnc_addWeaponWithoutItems;
+        [player, "arifle_AK12_lush_arco_snds_pointer_bipod_F", false] call CBA_fnc_addWeaponWithoutItems;
     (end)
 
 Author:
-    commy2
+    commy2, johnb43
 ---------------------------------------------------------------------------- */
 
-params ["_unit", "_weapon"];
+params ["_unit", "_weapon", ["_removeLinkedItems", true, [true]]];
 
 // config case
-private _compatibleMagazines = [_weapon, true] call CBA_fnc_compatibleMagazines;
+private _compatibleMagazines = compatibleMagazines _weapon;
```

Comment:
> Are we sure this reports magazines in config case?

From my testing, yes, it reports magazines in config case. I tested numerous weapons with multiple mods loaded (RHS, FOW, IFA, 3CB Factions & BAF) - however I can't *guarantee* it will always return config case, I can only strongly assume.

Reply:
Dedmen was able to confirm that both `compatibleMagazines` and `compatibleItems` return config sensitive strings.